ch_pipeline/analysis/flagging.py

```diff
@@ -1702,6 +1702,348 @@ def process(self, ss):
         return ss
 
 
+class MaskDecorrelatedCylinder(task.SingleTask):
+    """Identify and mask frequencies and times where a cylinder decorrelated.
+
```

Comment:
I think it might be useful for others to just have a few sentences describing the general algorithm in here. It is well commented throughout, but it would help someone coming in with no prior knowledge orient themselves a little more.

Reply:
I've added comments that describe these decorrelated cylinder events and the algorithm used to identify them.